libs/langgraph/langgraph/prebuilt/sequential_executor.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Optional, Type, Union, cast
+
+from langchain_core.runnables.base import Runnable, RunnableLike
+
+from langgraph.checkpoint.base import BaseCheckpointSaver
+from langgraph.graph.state import END, START, CompiledStateGraph, StateGraph
+from langgraph.store.base import BaseStore
+
+
+def _get_name(step: RunnableLike) -> str:
+    if isinstance(step, Runnable):
+        if step.name is None:
+            raise ValueError(
+                f"Runnable ({step}) needs to have a name attribute. "
+                "Consider setting the name or passing it as a tuple (name, runnable)."
+            )
+        return step.name
+    elif callable(step):
+        return getattr(step, "__name__", step.__class__.__name__)
+    else:
+        raise TypeError(f"Unsupported step type: {step}")
+
+
+def create_sequential_executor(
+    *steps: Union[RunnableLike, tuple[str, RunnableLike]],
+    state_schema: Type[Any],
+    checkpointer: Optional[BaseCheckpointSaver] = None,
+    store: Optional[BaseStore] = None,
+    interrupt_before: Optional[list[str]] = None,
+    interrupt_after: Optional[list[str]] = None,
+    debug: bool = False,
+) -> CompiledStateGraph:
+    """Creates a sequential executor graph that runs a series of provided steps in order.
+
+    Args:
+        *steps: A sequence of RunnableLike objects or (name, RunnableLike) tuples.
+            If no names are provided, the name will be inferred from the step object (e.g. a runnable or a callable name).
+            Each step will be executed in the order provided.
+        state_schema: The state schema for the graph.
+        checkpointer: An optional checkpoint saver object. This is used for persisting
+            the state of the graph (e.g., as chat memory) for a single thread (e.g., a single conversation).
+        store: An optional store object. This is used for persisting data
+            across multiple threads (e.g., multiple conversations / users).
+        interrupt_before: An optional list of step names to interrupt before execution.
+        interrupt_after: An optional list of step names to interrupt after execution.
+        debug: A flag to enable debug mode.
+
+    Returns:
+        A CompiledStateGraph object.
+    """
+    if len(steps) < 2:
+        raise ValueError("Sequential executor requires at least two steps.")
+
+    builder = StateGraph(state_schema)
+    previous_name: Optional[str] = None
+    for step in steps:
+        if isinstance(step, tuple) and len(step) == 2:
+            name, step = step
+        else:
+            name = _get_name(step)
+
+        builder.add_node(name, step)
+        if previous_name is None:
+            builder.add_edge(START, name)
+        else:
+            builder.add_edge(previous_name, name)
+
+        previous_name = name
+
+    builder.add_edge(cast(str, previous_name), END)
+    return builder.compile(
```

Comment:
> Should it be compiled or uncompiled?

Also not sure, had the same question. I think might be nice to have an option to return either (even though it's a bit icky to have multiple return types like this). But i think this needs to be consistent with `create_react_agent`

> Can it be added a sub-graph after being compiled?

it can definitely be added as a subgraph after being compiled :) you actually can't add it uncompiled (can either add a compiled subgraph as a node or invoke the compiled subgraph inside a node func)

> Will users be confused about how to interpret where to provide a checkpointer?

why do you think so? think should just document it well. it would probably be more confusing if we have options to return either compiled or uncompiled

Reply:
I still feel strongly that it should be uncompiled

Reply:
added an option to return either, might need to add the same to `create_react_agent`

Reply:
I personally think it should always be uncompiled, especially for this "lower-level" type method
and that we made the wrong decision for create_react_agent

Reply:
sounds good! i kept is uncompiled in the current version of the PR